controllers/managedocs_controller.go

```diff
@@ -255,3 +323,104 @@ func (r *ManagedOCSReconciler) updateComponents() v1.ComponentStatusMap {
+		Name:      r.DeleteConfigMapName,
+		Namespace: r.namespace,
+	}
+
+	configmap := &corev1.ConfigMap{}
+	err := r.Get(r.ctx, cmNamespaceName, configmap)
+	if err == nil {
+		labels := configmap.Labels
+		if _, ok := labels[r.DeleteConfigMapLabelKey]; !ok {
+			return false
+		}
+		return true
+	}
+	return false
+}
+
+func (r *ManagedOCSReconciler) removeOlmComponents() (reconcile.Result, error) {
+
+	r.Log.Info("Deleting subscription")
+	if result, err := r.deleteResource(r.Client, &operators.Subscription{
+		ObjectMeta: metav1.ObjectMeta{Name: r.AddonSubscriptionName, Namespace: r.namespace},
+	}, r.AddonSubscriptionName); err != nil {
+		return ctrl.Result{}, err
+	} else if !result.IsZero() {
+		return result, nil
+	}
+	r.Log.Info("subscription removed successfully")
+
+	r.Log.Info("Deleting CSV")
+	csvList := &operators.ClusterServiceVersionList{}
+	err := r.Client.List(context.Background(), csvList)
+	if err != nil {
+		r.Log.Error(err, "Unable to list CSVs")
+		return ctrl.Result{}, err
+	}
+	for _, csv := range csvList.Items {
+		if strings.HasPrefix(csv.Name, deployerCsvPrefix) {
+			deployerCsvName := csv.Name
+			if result, err := r.deleteResource(r.Client, &operators.ClusterServiceVersion{
+				ObjectMeta: metav1.ObjectMeta{Name: deployerCsvName, Namespace: r.namespace},
+			}, deployerCsvName); err != nil {
+				return ctrl.Result{}, err
+			} else if !result.IsZero() {
+				return result, nil
+			}
+		}
+	}
+	r.Log.Info("CSV removed successfully")
+
+	return ctrl.Result{}, nil
+}
+
+func (r *ManagedOCSReconciler) isComponentReady() bool {
+	subComponent := r.managedOCS.Status.Components
+
+	if subComponent.StorageCluster.State != v1.ComponentReady {
+		return false
+	}
+	return true
+}
+
+func (r *ManagedOCSReconciler) isPvcRemoved() (bool, error) {
+	pvcList := &corev1.PersistentVolumeClaimList{}
+	err := r.UnrestrictedClient.List(context.Background(), pvcList)
+	if err != nil {
+		r.Log.Error(err, "Unable to list PVCs")
+		return false, err
+	}
+	for _, pvc := range pvcList.Items {
+		scName := *pvc.Spec.StorageClassName
+		if scName == storageClassCephFsName || scName == storageClassRbdName {
+			r.Log.Info("Found Consumer PVCs using OCS storageclasses, cannot proceed on uninstalltion")
+			return false, nil
+		}
+	}
+	return true, nil
+}
+
+func (r *ManagedOCSReconciler) deleteResource(cl client.Client, resource runtime.Object, resourceName string) (reconcile.Result, error) {
+	err := cl.Delete(context.Background(), resource, client.PropagationPolicy(metav1.DeletePropagationForeground))
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		r.Log.Error(err, "Uninstall: unable to delete resource", "resource", resourceName)
+		return ctrl.Result{}, err
+	}
+	err = cl.Get(context.Background(), types.NamespacedName{Name: resourceName,
+		Namespace: r.namespace}, resource)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, err
+	}
```

Comment:
Why do we need this? If sync between deletes is needed the check should be a condition for the next delete not a condition for completion on the current delete.

Reply:
removing it based on the other changes, as it not required in our case .